projects/cdk/pipes/replace/replace.pipe.ts

```diff
@@ -0,0 +1,16 @@
+import {Pipe, PipeTransform} from '@angular/core';
+
+type Replacer = (substring: string, ...args: any[]) => string;
+
+@Pipe({name: `tuiReplace`})
+export class TuiReplacePipe implements PipeTransform {
+    transform(
+        value: string,
+        search: RegExp | string,
+        replacer: Replacer | string,
```

Comment:
Is it some kind of typescript issue? Can we make it extract type automatically from `String.prototype.replace`? Also let's make `value: string | null` to support `async` pipes since we check for falsy values anyway. BTW, I don't think it makes sense to check for falsy `search` and `replacer`, who would use a pipe like that? Just `return value && value.replace(...)` is fine, IMO.

Reply:
`extract type automatically from String.prototype.replace` it doesn't work
https://github.com/microsoft/TypeScript/issues/22378

`return value && value.replace(...) is fine, IMO.` updated

Reply:
fixed